app/models/manageiq/providers/lenovo/physical_infra_manager/physical_server.rb

```diff
@@ -2,6 +2,12 @@ module ManageIQ::Providers
   class Lenovo::PhysicalInfraManager::PhysicalServer < ::PhysicalServer
     include_concern 'RemoteConsole'
 
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :physical_servers,
+               :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager"
+
+    belongs_to :physical_rack, :dependent => :destroy, :inverse_of => :physical_servers,
+               :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalRack"
```

Comment:
Again I think all of these associations should be defined on the base models without the class_name, not here